annotation-server/src/guidelines/guidelines.service.ts

```diff
@@ -0,0 +1,322 @@
+    }
+
+    async fetchGuidelines(): Promise<void> {
+        this.clearAllData();
+        const guidelines = await this.fetchCpicGuidelines();
+        await this.complementAndSaveGuidelines(guidelines);
+    }
+
+    async complementAndSaveGuidelines(
+        guidelines: Map<string, Guideline[]>,
+    ): Promise<void> {
+        const [
+            medications,
+            genes,
+            phenotypeHeader,
+            implications,
+            recommendations,
+        ] = await fetchSpreadsheetCells(
+            this.configService.get<string>('GOOGLESHEET_ID'),
+            this.configService.get<string>('GOOGLESHEET_APIKEY'),
+            [
+                this.configService.get<string>('GOOGLESHEET_RANGE_MEDICATIONS'),
+                this.configService.get<string>('GOOGLESHEET_RANGE_GENES'),
+                this.configService.get<string>('GOOGLESHEET_RANGE_PHENOTYPES'),
+                this.configService.get<string>(
+                    'GOOGLESHEET_RANGE_IMPLICATIONS',
+                ),
+                this.configService.get<string>(
+                    'GOOGLESHEET_RANGE_RECOMMENDATIONS',
+                ),
+            ],
+        );
+
+        this.spreadsheetPhenotypeHeader = phenotypeHeader[0].map(
+            (cell) =>
+                new Set(
+                    cell.value
+                        .split(';')
+                        .map((phenotype) => phenotype.trim().toLowerCase()),
+                ),
+        );
+        for (let row = 0; row < medications.length; row++) {
+            const geneSymbolName = genes[row]?.[0];
+            const medicationName = medications[row]?.[0];
+            if (!geneSymbolName || !medicationName) continue;
+
+            const medication = await this.findMedicationByName(
+                medicationName.value,
+            );
+            const genePhenotypes = await this.findGenePhenotypes(
+                geneSymbolName.value,
+            );
+            if (genePhenotypes.length === 0 || !medication) continue;
+
+            for (let col = 0; col < implications[row].length; col++) {
+                for (const genePhenotype of genePhenotypes[col].values()) {
+                    const guidelinesForMedication = guidelines.get(
+                        medication.name,
+                    );
+                    const guidelinesForGenePhenotype =
+                        guidelinesForMedication.filter(
+                            (guidelineForMed) =>
+                                guidelineForMed.genePhenotype.id ===
+                                genePhenotype.id,
+                        );
+                    if (!guidelinesForGenePhenotype.length)
+                        throw new InternalServerErrorException(
+                            `No matching CPIC guideline was found for ${medication.name} and genephenotype ${genePhenotype.geneSymbol}, ${genePhenotype.phenotype.name}!`,
+                        );
```

Comment:
Is there a reason we handle this so differently from the *couldn't find medication* errors below? I think finding a common approach for all the errors might make sense

Reply:
This is out of scope for this PR, opened #256